trade-finance-manager-api/src/v1/helpers/amendment.helpers.js

```diff
@@ -515,13 +514,35 @@ const addLatestAmendmentDates = async (tfmObject, latestCoverEndDateResponse, la
 
   const coverEndDate = latestCoverEndDateResponse?.coverEndDate;
   const amendmentExposurePeriodInMonths = coverEndDate ? await calculateAmendmentDateTenor(coverEndDate, existingFacility) : undefined;
-  const facilityEndDate = latestFacilityEndDateDataResponse?.facilityEndDate;
 
   return {
     ...tfmObject,
     coverEndDate,
     amendmentExposurePeriodInMonths,
-    facilityEndDate,
+  };
+};
+
+/**
+ * Populates the tfmObject with facility end date values
+ * @param {import('@ukef/dtfs2-common').TfmFacility} tfmObject
+ * @param {{ facilityEndDate?: string, bankReviewDate?:string, isUsingFacilityEndDate?: boolean } | undefined} latestFacilityEndDateDataResponse
+ * @param {string} facilityId
+ * @returns {import('@ukef/dtfs2-common').TfmFacility & { coverEndDate?: string, amendmentExposurePeriodInMonths?: number, facilityEndDate?: string }}
+ */
+const addLatestAmendmentFacilityEndDate = async (tfmObject, latestFacilityEndDateDataResponse, facilityId) => {
+  const existingFacility = await api.findOneFacility(facilityId);
+
+  if (!existingFacility) {
+    return tfmObject;
+  }
+
+  const { isUsingFacilityEndDate, facilityEndDate, bankReviewDate } = Object(latestFacilityEndDateDataResponse);
```

Comment:
`latestFacilityEndDateDataResponse` could be undefined, so by doing `Object` on it, it would instead access the values on `{}` which would stop it from erroring. However, have refactored to check in the lines above that `latestFacilityEndDateDataResponse` isn't undefined :)